patroni/config_generator.py

```diff
@@ -0,0 +1,266 @@
+        if cur.rowcount < 1:
+            sys.exit('The provided user does not have superuser privilege')
+
+        required_params = ['hba_file', 'ident_file', 'config_file',
+                           'data_directory'] + list(ConfigHandler.CMDLINE_OPTIONS.keys())
+        cur.execute("SELECT name, current_setting(name) FROM pg_settings \
+                     WHERE context <> 'internal' \
+                     AND source IN ('configuration file', 'command line', 'environment variable') \
+                     AND category <> 'Write-Ahead Log / Recovery Target' \
+                     AND setting <> '(disabled)' \
+                     OR name = ANY(%s);", (required_params,))
+
+        helper_dict = dict.fromkeys(['port', 'listen_addresses'])
+        # adjust values
+        for p, v in cur.fetchall():
+            if p == 'data_directory':
+                config['postgresql']['data_dir'] = v
+            elif p == 'cluster_name' and v:
+                config['scope'] = v
+            elif p in ('archive_command', 'restore_command', 'archive_cleanup_command',
+                       'recovery_end_command', 'ssl_passphrase_command',
+                       'hba_file', 'ident_file', 'config_file'):
+                # write commands to the local config due to security implications
+                # write hba/ident/config_file to local config to ensure they are not removed later
+                config['postgresql'].setdefault('parameters', {})
+                config['postgresql']['parameters'][p] = v
+            elif p in helper_dict:
+                helper_dict[p] = v
+            else:
+                config['bootstrap']['dcs']['postgresql']['parameters'][p] = v
+
+    conn.close()
+
+    connect_port = parsed_dsn.get('port', os.getenv('PGPORT', helper_dict['port']))
+    config['postgresql']['connect_address'] = f'{get_ip()}:{connect_port}'
+    config['postgresql']['listen'] = f'{helper_dict["listen_addresses"]}:{helper_dict["port"]}'
+
+    # it makes sense to define postgresql.pg_hba/pg_ident only if hba_file/ident_file are set to defaults
+    default_hba_path = os.path.join(config['postgresql']['data_dir'], 'pg_hba.conf')
+    if config['postgresql']['parameters']['hba_file'] == default_hba_path:
+        try:
+            with open(default_hba_path, 'r') as f:
+                config['postgresql']['pg_hba'] = list(filter(lambda i: i and i.split()[0] in ('local',
+                                                                                              'host',
+                                                                                              'hostssl',
+                                                                                              'hostnossl',
+                                                                                              'hostgssenc',
+                                                                                              'hostnogssenc'),
```

Comment:
v16 will support different kinds of "include": https://www.postgresql.org/docs/16/release-16.html#RELEASE-16-PG-HBA

Reply:
did I get right what you mean?:)
https://github.com/hughcapet/patroni/pull/2/commits/ebfcfa53fee19d8f7360ee59b1a738b0a17abe82

or we want to allow those lines regardless pg version?

Reply:
Regardless of the version would also be totally fine.